starknet_py/utils/typed_data.py

```diff
@@ -164,12 +168,35 @@ def _encode_data(self, type_name: str, data: dict) -> List[int]:
         return values
 
     def _verify_types(self):
-        reserved_type_names = ["felt", "felt*", "string", "selector", "merkletree"]
+        if self.domain.separator_name not in self.types:
+            raise ValueError(f"Types must contain '{self.domain.separator_name}'.")
+
+        reserved_type_names = ["felt", "string", "selector", "merkletree"]
 
         for type_name in reserved_type_names:
             if type_name in self.types:
                 raise ValueError(f"Reserved type name: {type_name}")
 
+        referenced_types = [
+            parameter for type_name in self.types for parameter in self.types[type_name]
+        ]
+        referenced_types = [
+            ref_type.contains
+            if ref_type.contains is not None
+            else strip_pointer(ref_type.type)
+            for ref_type in referenced_types
+        ] + [self.domain.separator_name, self.primary_type]
```

Comment:
nit: I think it does the same thing, but without duplicates and readability seems to be slightly better.
```suggestion
        referenced_types = {
            ref_type.contains
            if ref_type.contains is not None
            else strip_pointer(ref_type.type)
            for type_name in self.types
            for ref_type in self.types[type_name]
        }

        referenced_types.update([self.domain.separator_name, self.primary_type])
```